File: test_supa.py
import urllib.request
import json

url = "https://ajkhkmrxwfpqwywujhfu.supabase.co/rest/v1/Client?select=*&limit=1"
headers = {
    "apikey": "sb_publishable_N0TtoWxyljFXRp3boJV1tg_o5yoliT8",
    "Authorization": "Bearer sb_publishable_N0TtoWxyljFXRp3boJV1tg_o5yoliT8",
    "Origin": "http://localhost:8000"
}

req = urllib.request.Request(url, headers=headers)
try:
    with urllib.request.urlopen(req) as response:
        print("Status:", response.status)
        print("Headers:", response.headers)
        print("Data:", response.read().decode())
except Exception as e:
    print("Error:", e)

File: test_supa.js
const { createClient } = require('@supabase/supabase-js');

const supabaseUrl = 'https://ajkhkmrxwfpqwywujhfu.supabase.co';
const supabaseKey = 'sb_publishable_N0TtoWxyljFXRp3boJV1tg_o5yoliT8';
const supabase = createClient(supabaseUrl, supabaseKey);

async function test() {
    console.log("Testing connection...");
    const { data, error } = await supabase.from('Client').select('*').limit(1);
    
    if (error) {
        console.error("Error:", error);
    } else {
        console.log("Success data:", data);
        
        // Let's insert dummy
        const testData = {
                id: '123e4567-e89b-12d3-a456-426614174000',
                "Client Name": "Test Client Insert From Node",
        };
        const {data: insertData, error: insertError} = await supabase.from('Client').upsert([testData]);
        if(insertError) {
             console.error("Insert error:", insertError.message, insertError.details, insertError.hint);
        } else {
             console.log("Insert success");
        }
    }
}

test();
